src/SignerAlgorithm.ts

```diff
@@ -1,49 +1,25 @@
-import { Signer, SignerAlgorithm } from './JWT'
-import { EcdsaSignature, fromJose, toJose } from './util'
+import { SignerAlgorithm } from './JWT'
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-function instanceOfEcdsaSignature(object: any): object is EcdsaSignature {
-  return typeof object === 'object' && 'r' in object && 's' in object
-}
-
-export function ES256KSignerAlg(recoverable?: boolean): SignerAlgorithm {
-  return async function sign(payload: string, signer: Signer): Promise<string> {
-    const signature: EcdsaSignature | string = await signer(payload)
-    if (instanceOfEcdsaSignature(signature)) {
-      return toJose(signature, recoverable)
-    } else {
-      if (recoverable && typeof fromJose(signature).recoveryParam === 'undefined') {
-        throw new Error(`not_supported: ES256K-R not supported when signer doesn't provide a recovery param`)
-      }
-      return signature
-    }
-  }
-}
+import * as ES256KSignerAlg from './SignerAlg/ES256KSignerAlg'
+import * as Ed25519SignerAlg from './SignerAlg/Ed25519SignerAlg'
 
-export function Ed25519SignerAlg(): SignerAlgorithm {
-  return async function sign(payload: string, signer: Signer): Promise<string> {
-    const signature: EcdsaSignature | string = await signer(payload)
-    if (!instanceOfEcdsaSignature(signature)) {
-      return signature
-    } else {
-      throw new Error('invalid_config: expected a signer function that returns a string instead of signature object')
-    }
-  }
-}
+import * as ES256SignerAlg from './SignerAlg/ES256SignerAlg'
 
 interface SignerAlgorithms {
   [alg: string]: SignerAlgorithm
 }
 
 const algorithms: SignerAlgorithms = {
-  ES256K: ES256KSignerAlg(),
+  ES256: ES256SignerAlg.ES256SignerAlg(),
+  'ES256-R': ES256SignerAlg.ES256SignerAlg(true),
+  ES256K: ES256KSignerAlg.ES256KSignerAlg(),
```

Comment:
With the above suggestion, this would simplify to:
```suggestion
  ES256K: ES256KSignerAlg(),
```